crates/pgt_workspace/src/workspace/server/parsed_document.rs

```diff
@@ -404,3 +404,27 @@ impl<'a> StatementFilter<'a> for IdFilter {
         *id == self.id
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use pgt_fs::PgTPath;
+
+    #[test]
+    fn sql_function_body() {
+        let input = "CREATE FUNCTION add(integer, integer) RETURNS integer
+    AS 'select $1 + $2;'
+    LANGUAGE SQL
+    IMMUTABLE
+    RETURNS NULL ON NULL INPUT;";
+
+        let path = PgTPath::new("test.sql");
+
+        let d = ParsedDocument::new(path, input.to_string(), 0);
+
+        let stmts = d.iter(DefaultMapper).collect::<Vec<_>>();
+
+        assert_eq!(stmts.len(), 2);
```

Comment:
Love it!

should we add a test asserting the content as well? 
and maybe a test using the $$ notation?

Reply:
not really necessary imo - we can trust pg_query here

Reply:
will do it in a follow up